ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -1306,6 +1308,122 @@ public void shouldFailTerminateOnInvalidDeleteTopicPattern() {
     ksqlResource.terminateCluster(request);
   }
 
+  @Test
+  public void shouldFailIfRegisterTopicAlreadyExists() {
+    // Given:
+    final String registerSql = "REGISTER TOPIC FOO WITH (kafka_topic='bar', value_format='json');";
+    metaStore.putTopic(new KsqlTopic("FOO", "foo", new KsqlJsonTopicSerDe(), true));
+
+    when(commandStore.enqueueCommand(any(), any(), any(), any()))
```

Comment:
It's a little strange to have a statement like this in the test.  I generally find it best to have each test only test one thing. Maybe this would be better as two tests, e.g. `shouldThrowIfRegisterTopicAlreadyExists` and `shouldNotEnqueueCommandIfTopicAlreadyExists`.